.github/actions/sync-issue-with-jira/action.yaml

```
@@ -0,0 +1,120 @@
+name: "Sync GitHub issue to Jira"
+description: 'This GitHub action creates Jira issue from GitHub issue when GitHub issue is labeled with "jira" label.'
+inputs:
+  webhook-url:
+    description: >
+      Jira integration webhook URL.
+      Store it as a secret as anyone who has access to it will be able to post to your Jira board.
+    required: true
+  label:
+    description: "Label which will trigger a Jira import."
+    required: true
+    default: "jira"
+
+runs:
+  using: "composite"
+  steps:
+    - name: restrict action to labelled issues
+      run: |
+        set -eux
+
+        echo "NeedsJiraUpdate=false" >> $GITHUB_ENV
+
+        if [ ${{ github.event_name }} != "issues" ]; then
+          echo "This action only work on issue events. Please use on: issues to use this action."
+          exit 1
+        fi
+
+        if [ ${{ github.event.issue.pull_request }} ]; then
+          echo "This action only work on issues, not pull requests."
+          exit 0
+        fi
+
+        # Issue creation with label will trigger 2 events and run twice: one create, one labelled.
+        # let just focus on labelling then for creating issues Jira-side.
+        if [ ${{ github.event_name }} == "issues" ] && [ ${{ github.event.action }} == "opened" ]; then
+          echo "Ignoring creation of issues as a label will trigger a second event."
+          exit 0
+        fi
+
+        # We only operate on labelled issues or issues that are just unlabeled with our desired label
+        ## check if one label of labels is our jira label
+        toconsider=${{ contains(github.event.issue.labels.*.name, inputs.label) }}
+        ## second chance, this has just been unlabeled and needs to be deleted on Jira
+        if [ ${{ github.event.action }} == "unlabeled" ] && [ ${{ github.event.label.name }} == ${{ inputs.label }} ]; then
+          toconsider=true
+        fi
+        if [ "${toconsider}" == false ]; then
+          echo "Our desired label not found on issue or not unlabeled, skipping"
+          exit 0
+        fi
+
+        # And finally, for the "labeled" event, we are only interested if the new added label is our desired one.
+        if [ ${{ github.event.action }} == "labeled" ] && [ ${{ github.event.label.name }} != ${{ inputs.label }} ]; then
+          echo "Not interested in this action, skipping"
+          exit 0
+        fi
+
+        # last one wins
+        echo "NeedsJiraUpdate=true" >> $GITHUB_ENV
+      shell: bash
+
+    - name: "Update jira"
+      if: ${{ env.NeedsJiraUpdate == 'true' }}
```

can we use something like this instead of the whole script step above?  if so, we should also preserve some of your comments there, especially the "Issue creation with label will trigger 2 events...".
```suggestion
      if: |
        github.event_name == "issues"
        && ! github.event.issue.pull_request
        && (contains(github.event.issue.labels.*.name, inputs.label) || github.event.label.name == inputs.label)
        && (github.event.action == "labeled" || github.event.action == "unlabeled" || github.event.action == "reopened" || github.event.action == "closed")
        && ( !(github.event.action == "labeled" || github.event.action == "unlabeled") || github.event.label.name == inputs.label )
        # line by line breakdown of above:
        #  - event is "issues"
        #  - and its not a pull request
        #  - and it has the keyword among its existing labels, OR this is an actual keyword [un]label action itself
        #  - and its action is one of "labeled", "unlabeled", "reopened", or "closed"
        #  - and if its action is "labeled" or "unlabeled", the label name matches the keyword
```

---

I think so, I will test that suggestion and if it works I will add those changes